products/monitoring/terraform.yaml

```diff
@@ -0,0 +1,97 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Provider::ResourceOverrides
+  AlertPolicy: !ruby/object:Provider::Terraform::ResourceOverride
+    id_format: "{{name}}"
+    mutex: alertPolicy/{{project}}
+    examples: |
+      ### Basic Usage
+      ```hcl
+      resource "google_monitoring_alert_policy" "basic" {
+        display_name = "Test Policy Basic"
+        combiner = "OR"
+        conditions = [
+          {
+            display_name = "test condition"
+            condition_threshold {
+              filter = "metric.type=\"compute.googleapis.com/instance/disk/write_bytes_count\" AND resource.type=\"gce_instance\""
+              duration = "60s"
+              comparison = "COMPARISON_GT"
+              aggregations = [
+                {
+                  alignment_period = "60s"
+                  per_series_aligner = "ALIGN_RATE"
+                }
+              ]
+            }
+          }
+        ]
+      }
+      ```
+    properties:
+    # The upstream API expects name to be `projects/[PROJECT_ID]/alertPolicies/[POLICY_ID]` but
+    # we are storing just the {POLICY_ID} as the id/name in terraform state. This is because 
+    # import logic constructs a url from the id as isn't currently tolerant of id's that are
```

Comment:
That's not true though, right? The import fn itself doesn't call get, only the read does, and it does it based on the self_link_url set in api.yaml, which in this case would have been {{name}}. The import fn would have just set name in state (you should be able to override the list of allowed import values to make it only be {{name}} or write a custom import function). The id itself is never read anywhere.

I want to make sure we get this right now, since I think this sort of thing is going to pop up a lot in the future (https://cloud.google.com/apis/design/standard_fields) and I want to make sure we maintain the ability for people to correctly interpolate based on fields that are in the API. If another resource needed to refer to an AlertPolicy, I assume it would do so based on its full name (unfortunately stackdriver doesn't have any examples on this, so I don't know for sure), which means we want to store that full name _somewhere_, and if we can have it just match the API that would be really nice.

Reply:
I might have been muddling things, but I'll see if I can explain a bit more: At this point we'd need to rewrite the regex generation (more details below) to use `import_format`. I did forget that the Read method is called with all the context still set on the resource so I was thinking that the id was being used but you're right that it's not actually being used.

re:regex - Changing the `import_format` to `{{name}} with the way the regex is currently built will translate into regex of `(?P<name>[^/]+)` where it's getting tripped up by the fact there is a slash in the name and not matching the whole id.

I hadn't thought about just using a custom import, so that would probably be a much cleaner option then what I have going so far. Probably the 'right' way would be to fix the regex, but I'm pretty sure it would take changing the id_format input significantly or introducing something like `id_regex` so I was hesitant to do that without discussion strategy in advance.